components/movies/MovieDetails.bs

```diff
@@ -568,20 +568,28 @@ sub SetUpAudioOptions(streams)
 end sub
 
 sub SetDefaultAudioTrack(itemData)
-    firstAudioTrackName = string.EMPTY
-    preferredAudioTrackIndex = m.global.queueManager.callFunc("getPreferredAudioTrackIndex") + 1
+    preferredAudioTrackIndex = m.global.queueManager.callFunc("getPreferredAudioTrackIndex")
+    preferredLanguage = m.global.session.user.Configuration.AudioLanguagePreference
+    playDefault = m.global.session.user.Configuration.PlayDefaultAudioTrack
+
+    ' Do we have a preferred audio track?
+    if preferredAudioTrackIndex < 0 and not isValid(playDefault)
+        preferredAudioTrackIndex = findDefaultTrack(itemData.mediaStreams)
+    end if
+
+    ' No selction, but they have Default as their preference?
+    if isValid(playDefault) and playDefault
```

Comment:
The comment above states there is no selection - meaning the user didn't manually select a preferred audio track - however, this condition is only checking playDefault. Doesn't it also need to check the `m.global.queueManager.callFunc("getPreferredAudioTrackIndex")` value?

Reply:
Line 575 right above it.